### PR TITLE
Configurable expoM for Rate Mode

### PR DESCRIPTION
**Pull request by fujin**
- Add expoM functionality/proto to misc_math supplied by @mlyle;
  allows the  user to specify the exponent for the curve generation.
- Add expoM calls to transmitter control for rate axis
- UAVo for RateExponent. Maxes at 25 for now.
- [ ] inverse expoM for horizon GCS

References TauLabs/TauLabs#1785
Closes d-ronin/dRonin#108

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/109)

### CODE REVIEW

File: flight/Libraries/math/misc_math.c [hunk 1]
```diff
@@ -115,6 +115,20 @@ float expo3(float x, int32_t g)
...
 /**
+ * Approximate a configurable exponential scale curve.
+ * @param[in] input       Input to scale (from [-1,1])
+ * @param[in] exponent    The exponent passed to powf, or inverted and called recursively 
+ * @param[in] amount      Sets the exponential amount [0,100]
+ * @return  rescaled input
+ */
+float expoM(float input, float exponent, int32_t amount)
+{
+  if (input < 0)
```
Comment (mlyle) [inline flight/Libraries/math/misc_math.c]: indent

Comment (fujin) [inline flight/Libraries/math/misc_math.c]: fixed @ 81a7987

**Comment by fujin** [comment]
@mlyle you mentioned having an idea for inverse-expoM in the past, to help with horizon gcs? pending flight test, are you still happy to submit that?

**Comment by mlyle** [comment]
I can invert the function but there's not a closed form solution.  It's a little bit of work but I'll get to it eventually.

**Comment by mlyle** [comment]
recommendations for early test scenario initial tuning:

high deg/s (1200? 1400?)
pitch/roll exponents = 7
85% "expo" pitch/roll

Graphs in the GCS will not be accurate.

**Comment by mluessi** [comment]
@fujin can you test this? Your are probably the most qualified person to judge how it feels ;). Before merging, we should update the GCS code so the graphs are accurate.

**Comment by fujin** [comment]
@mlyle can you provide a recommendation around a maximum "accelerated" roll rate at extents around 900? I usually configure 720 roll, and see max of about 900~950 with 40% insanityfactor.

@mluessi of course, I'll fight test! we need the invert expo-M function from @mlyle (or to build that) to make the GCS horizon code work if I recall correctly.

**Comment by mlyle** [comment]
I think if you configure 1200 you won't really hit 1200.  You could pick just a little lower deg/s like 1000 and 80% expo instead.

File: flight/Libraries/math/misc_math.c [hunk 2]
```diff
@@ -115,6 +115,20 @@ float expo3(float x, int32_t g)
...
+ * @param[in] input       Input to scale (from [-1,1])
+ * @param[in] exponent    The exponent passed to powf, or inverted and called recursively 
+ * @param[in] amount      Sets the exponential amount [0,100]
+ * @return  rescaled input
+ */
+float expoM(float input, float exponent, int32_t amount)
+{
+	if (input < 0)
+		return -expoM(-input, exponent, amount);
+	return (input * ((100 - amount) / 100.0f) + powf(input, exponent) * (amount / 100.0f));
```
Comment (mluessi) [inline flight/Libraries/math/misc_math.c:128]: we could probably use `fastpow` here

**Comment by mlyle** [comment]
Superseded by #1250 (WIP)

**Pull request closed without merging**